pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,244 @@
+			NodeSelector:                  c.config.EgressGW.GetNodeSelector(),
+			TerminationGracePeriodSeconds: c.config.EgressGW.GetTerminationGracePeriod(),
+			InitContainers:                []corev1.Container{*c.egwInitContainer()},
+			Containers:                    []corev1.Container{*c.egwContainer()},
+			Volumes:                       []corev1.Volume{*c.egwVolume()},
+		},
+	}
+}
+
+func (c *component) egwBuildAnnotations() map[string]string {
+	annotations := map[string]string{}
+	annotations["cni.projectcalico.org/ipv4pools"] = c.config.EgressGW.GetIPPools()
+	if c.config.EgressGW.Spec.AWS != nil && len(c.config.EgressGW.Spec.AWS.ElasticIPs) > 0 {
+		annotations["cni.projectcalico.org/awsElasticIPs"] = c.config.EgressGW.GetElasticIPs()
+	}
+	return annotations
+}
+
+func (c *component) egwInitContainer() *corev1.Container {
+	initContainerPrivileges := true
+	return &corev1.Container{
+		Name:            "egress-gateway-init",
+		Image:           c.config.egwImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command:         []string{"/init-gateway.sh"},
+		SecurityContext: &corev1.SecurityContext{Privileged: &initContainerPrivileges},
+		Env:             c.egwInitEnvVars(),
+	}
+}
+
+func (c *component) egwContainer() *corev1.Container {
+	return &corev1.Container{
+		Name:            "egress-gateway",
+		Image:           c.config.egwImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Env:             c.egwEnvVars(),
+		Resources:       c.egwResources(),
+		VolumeMounts:    c.egwVolumeMounts(),
+		Ports:           c.egwPorts(),
+		Command:         []string{"/start-gateway.sh"},
+		ReadinessProbe:  c.egwReadinessProbe(),
+		SecurityContext: c.egwSecurityContext(),
+	}
+}
+
+func (c *component) egwReadinessProbe() *corev1.Probe {
+	return &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/readiness",
+				Port: intstr.FromInt(int(DefaultHealthPort)),
+			},
+		},
+		InitialDelaySeconds: 3,
+		TimeoutSeconds:      1,
+		SuccessThreshold:    1,
+		PeriodSeconds:       3,
+	}
+}
+
+func (c *component) egwSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"NET_ADMIN"},
+		},
+	}
+}
+
+func (c *component) egwPorts() []corev1.ContainerPort {
+	return []corev1.ContainerPort{
+		{
+			ContainerPort: DefaultHealthPort,
+			Name:          EGWPortName,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+}
+
+func (c *component) egwResources() corev1.ResourceRequirements {
+	return c.config.EgressGW.GetResources()
+}
+
+func (c *component) egwVolume() *corev1.Volume {
+	return &corev1.Volume{
+		Name:         "policysync",
+		VolumeSource: corev1.VolumeSource{CSI: &corev1.CSIVolumeSource{Driver: "csi.tigera.io"}},
+	}
+}
+
+func (c *component) egwVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		corev1.VolumeMount{Name: "policysync", MountPath: "/var/run"},
```

Comment:
Did we always mount to `/var/run` ?  I'd have expected something like `/var/run/calico`  Perhaps now would be a good time to change that (and update the egressd arg to match).